File: client-app/src/interfaces/Comment.ts
export interface ICommentDisplay {
  id: number;
  parentId: number;
  content: string;
  createdAt: Date;
  updatedAt: Date;
  replyCount: number;
  userId: string;
  userDisplayName: string;
  userAvatarUrl: string | undefined;
}

File: Social/Social.Core/Entities/Comment.cs
namespace Social.Core.Entities;

public class Comment
{
    public int Id { get; set; }

    public int RecipeId { get; set; }

    public string UserId { get; set; } = string.Empty;

    public int? ParentId { get; set; }

    public string Path { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool IsDeleted { get; set; } = false;

    public string BuildChildrenPath()
    {
        return !string.IsNullOrEmpty(Path) ? $"{Path}.{Id}" : $"{Id}";
    }
}
